From 5ad4b795940428045ad87eb1ca42461b4f5de237 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 4 Mar 2020 21:25:46 +0100
Subject: [PATCH] src/utils/cst_tokenstream.c: initialize plast
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the following build failure:

src/utils/cst_tokenstream.c: In function ‘get_token_postpunctuation’:
src/utils/cst_tokenstream.c:732:51: error: ‘plast’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
         memmove(ts->postpunctuation, &ts->token[p + plast], (t - p));
                                                 ~~^~~~~~~

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 src/utils/cst_tokenstream.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/cst_tokenstream.c b/src/utils/cst_tokenstream.c
index 7c6ecdb..c3b27e5 100644
--- a/src/utils/cst_tokenstream.c
+++ b/src/utils/cst_tokenstream.c
@@ -705,7 +705,7 @@ static void get_token_sub_part_2(cst_tokenstream *ts,
 
 static void get_token_postpunctuation(cst_tokenstream *ts)
 {
-    int p, t, plast;
+    int p, t, plast = 0;
     const cst_string *one_cp;
 
     t = cst_strlen(ts->token);
